barretenberg/cpp/docs/src/sumcheck-outline.md

```diff
@@ -0,0 +1,438 @@
+
+- [Masking Evaluations of Multilinear Witnesses:](#MaskingEvalsOfWitnesses) 
+	At the stage of proving their evaluations at the challenge point, the witness polynomials fed to Sumcheck must not reveal any private information. 
+    We use a modification of Construction 3 described in <a href=" https://eprint.iacr.org/2019/317">Libra</a>  allowing the prover to open a new multilinear polynomial derived from the witnesses by adding a product of a random scalar and a public quadratic polynomial in the same number of variables.  
+
+- [Total Costs:](#ZKCosts) The effect of adding Libra technique and masking evaluations of multilinear witnesses on is assessed, and the theoretical upper bound on prover's work is compared to the implemenation costs. 
+
+Non ZK-Sumcheck Outline {#NonZKSumcheck}
+========
+- - -
+ ### Sumcheck Relation {#SumcheckRelation}
+
+ Given multilinear polynomials \f$ P_1,\ldots, P_N \in \mathbb{F}[X_0,\ldots, X_{d-1}] \f$ and a polynomial \f$ F \f$ in \f$ N \f$ variables, we run Sumcheck over the polynomial
+ \f{align}{
+    \tilde{F}
+    (X_0,\ldots, X_{d-1}) =
+    pow_{\beta}(X_0,\ldots, X_{d-1}) \cdot F\left( P_1 (X_0,\ldots, X_{d-1}), \ldots, P_N (X_0,\ldots, X_{d-1}) \right)
+    \f}
+to establish that \f$ F(P_1(\vec \ell),\ldots, P_N(\vec \ell) ) = 0 \f$, i.e. that \f$ F \f$ is satisfied at every
+point of \f$\{0,1\}^d\f$.
+
+ In the implementation, the relation polynomial \f$ F \f$ is specified by the Flavor.
+
+ ### Main Parameters {#MainParameters}
+
+The following constants are used in this exposition. 
+
+ |     Notation      |           | \f$ \sim \f$   Upper Bound |
+ --------------------|---------------|-----------|
+ | \f$ d \f$         | \ref multivariate_d "number of variables" in multilinear  polynomials \f$ P_1,\ldots, P_N\f$        | \f$ 20 \f$  |
+ | \f$ N \f$         | number of Prover Polynomials specified by <Flavor ::NUM_ALL_ENTITIES>                               | \f$ 60 \f$  | 
+ | \f$ N_w \f$  	 | number of Witness Polynomials specified bt <Flavor ::NUM_WITNESS_ENTITIES>                          | \f$ 17 \f$  | 
+ | \f$ n \f$         | \ref multivariate_n "size of the hypercube", i.e. \f$ 2^d\f$.                                       | \f$ 2^{20} \f$ |
+ | \f$ D \f$         | \ref bb::SumcheckProverRound< Flavor >::BATCHED_RELATION_PARTIAL_LENGTH "maximum partial degree of" \f$\tilde{F}\f$ | \f$ 12 \f$ |
+ | \f$ D_w\f$        | maximum witness degree of \f$ F \f$ | \f$ 5 \f$ |
+
+#### Maximum Witness Degree 
+The significance of this parameter becomes apparent in Section [Masking Evaluations of Multilinear Witnesses](#MaskingEvalsOfWitnesses). It is formally defined as follows 
+\f{align}{
+	D_w = \deg_{P_1, \ldots, P_{N_w}} F(P_1,\ldots, P_{N})
+\f}
+where by \f$ \deg_{P_1, \ldots, P_{N_w}} \f$ we mean the <b> total degree </b> of relation polynomial \f$ F \f$ in the first \f$N_w\f$ variables. 
```

Comment:
total is actually called the partial degree in code if I remember correctly (because total is only used in protogalaxy), and I guess it might make sense to explain here what partial degree is. Also I’m not quite sure what N_w variable means since N_w is the number of witness polynomials and they are all multivariates. Am I misunderstanding something?

Reply:
Tried to make it clearer. Also gave a small example below.